apis/keda/v1alpha1/triggerauthentication_webhook.go

```diff
@@ -0,0 +1,123 @@
+	return validateSpec(&ta.Spec)
+}
+
+func (ta *TriggerAuthentication) ValidateUpdate(old runtime.Object) (admission.Warnings, error) {
+	val, _ := json.MarshalIndent(ta, "", "  ")
+	scaledobjectlog.V(1).Info(fmt.Sprintf("validating triggerauthentication update for %s", string(val)))
+
+	oldTa := old.(*TriggerAuthentication)
+	if isTriggerAuthenticationRemovingFinalizer(ta.ObjectMeta, oldTa.ObjectMeta, ta.Spec, oldTa.Spec) {
+		triggerauthenticationlog.V(1).Info("finalizer removal, skipping validation")
+		return nil, nil
+	}
+	return validateSpec(&ta.Spec)
+}
+
+func (ta *TriggerAuthentication) ValidateDelete() (admission.Warnings, error) {
+	return nil, nil
+}
+
+// +kubebuilder:webhook:path=/validate-keda-sh-v1alpha1-clustertriggerauthentication,mutating=false,failurePolicy=ignore,sideEffects=None,groups=keda.sh,resources=clustertriggerauthentications,verbs=create;update,versions=v1alpha1,name=vsclustertriggerauthentication.kb.io,admissionReviewVersions=v1
+
+var _ webhook.Validator = &ClusterTriggerAuthentication{}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (cta *ClusterTriggerAuthentication) ValidateCreate() (admission.Warnings, error) {
+	val, _ := json.MarshalIndent(cta, "", "  ")
+	triggerauthenticationlog.Info(fmt.Sprintf("validating clustertriggerauthentication creation for %s", string(val)))
+	return validateSpec(&cta.Spec)
+}
+
+func (cta *ClusterTriggerAuthentication) ValidateUpdate(old runtime.Object) (admission.Warnings, error) {
+	val, _ := json.MarshalIndent(cta, "", "  ")
+	scaledobjectlog.V(1).Info(fmt.Sprintf("validating clustertriggerauthentication update for %s", string(val)))
+
+	oldCta := old.(*ClusterTriggerAuthentication)
+	if isTriggerAuthenticationRemovingFinalizer(cta.ObjectMeta, oldCta.ObjectMeta, cta.Spec, oldCta.Spec) {
+		triggerauthenticationlog.V(1).Info("finalizer removal, skipping validation")
+		return nil, nil
+	}
+
+	return validateSpec(&cta.Spec)
+}
+
+func (cta *ClusterTriggerAuthentication) ValidateDelete() (admission.Warnings, error) {
+	return nil, nil
+}
+
+func isTriggerAuthenticationRemovingFinalizer(om metav1.ObjectMeta, oldOm metav1.ObjectMeta, spec TriggerAuthenticationSpec, oldSpec TriggerAuthenticationSpec) bool {
+	taSpec, _ := json.MarshalIndent(spec, "", "  ")
+	oldTaSpec, _ := json.MarshalIndent(oldSpec, "", "  ")
+	taSpecString := string(taSpec)
+	oldTaSpecString := string(oldTaSpec)
+
+	return len(om.Finalizers) == 0 && len(oldOm.Finalizers) == 1 && taSpecString == oldTaSpecString
+}
+
+func validateSpec(spec *TriggerAuthenticationSpec) (admission.Warnings, error) {
+	if spec.PodIdentity != nil {
+		switch spec.PodIdentity.Provider {
+		case PodIdentityProviderAzure:
+			fallthrough
+		case PodIdentityProviderAzureWorkload:
+			if spec.PodIdentity.IdentityID != nil && *spec.PodIdentity.IdentityID == "" {
+				return nil, fmt.Errorf("identityid of PodIdentity should not be empty")
```

Comment:
As this will be the first error that users see, I'd be more clear, something like:
`if it's set, identityId has to be different than ""`
I mean, something clarifying that valid values are unset and other than ""